kuma/javascript/src/payments/routes.jsx

```diff
@@ -0,0 +1,67 @@
+// @flow
+import * as React from 'react';
+import Route, { type RouteComponentProps } from '../route.js';
+import Page from './pages/page.jsx';
+import LandingPage from './pages/index.jsx';
+import ThankYouPage from './pages/thank-you.jsx';
+import TermsPage from './pages/terms.jsx';
+
+type PaymentRoutesParams = {
+    locale: string,
+    slug: string,
+};
+
+export const PAYMENT_PATHS = {
+    THANK_YOU: 'thank-you',
+    TERMS: 'terms',
+};
+
+export function PaymentPage(props: RouteComponentProps) {
+    const { locale, slug, data } = props;
+
+    // remove forward slashes
+    const pathname = slug.replace(/\//g, '');
+
+    return (
+        <Page>
+            {{
+                [PAYMENT_PATHS.THANK_YOU]: <ThankYouPage locale={locale} />,
+                [PAYMENT_PATHS.TERMS]: <TermsPage data={data} />,
+            }[pathname] || <LandingPage data={data} locale={locale} />}
+        </Page>
+    );
+}
+
+// In order to use new URL() with relative URLs, we need an absolute base
+// URL. If we're running in the browser we can use our current page URL.
+// But if we're doing SSR, we just have to make something up.
+const BASEURL =
+    typeof window !== 'undefined' && window.location
+        ? window.location.origin
+        : 'http://ssr.hack';
+
+export class PaymentRoutes extends Route<PaymentRoutesParams, null> {
+    locale: string;
+
+    constructor(locale: string) {
+        super();
+        this.locale = locale;
+    }
+
+    getComponent() {
+        return PaymentPage;
+    }
+
+    match(url: string): ?PaymentRoutesParams {
+        const currentPath = new URL(url, BASEURL).pathname;
+        const paymentsPath = `/${this.locale}/payments`;
+
+        if (currentPath.startsWith(paymentsPath)) {
+            return {
+                locale: this.locale,
+                slug: currentPath.substring(paymentsPath.length),
```

Comment:
I guess it works but it feels a bit complicated. Is it perhaps because you have a need to export PAYMENT_PATHS to some sort of unit test?
The `PaymentPage` could be a simpler if something "dumber" like this:
```javascript
let page
if (url.endsWith('/thank-you')) {
  page = <ThankYouPage locale={locale} />
} else if (url.endsWith('/terms')) {
  page = <TermsPage data={data} />
} else {
  page = <LandingPage data={data} locale={locale} />
}
return <Page>
    {page}
  </Page>
```
(that might not exactly work but you get the point)

You don't *need* to change it but please consider it. These enums look a bit overly engineered.

Reply:
Thanks Peter, I can look into changing this in the subscription management work

Reply:
Never mind, I added these changes to this diff, except it's a switch statement since we'll be adding /management to this soon